src/client/Microsoft.Identity.Client/ManagedIdentity/ServiceFabricManagedIdentitySource.cs

```diff
@@ -48,9 +48,30 @@ public static AbstractManagedIdentity TryCreate(RequestContext requestContext)
             }
 
             requestContext.Logger.Verbose(() => "[Managed Identity] Creating Service Fabric managed identity. Endpoint URI: " + identityEndpoint);
+            ValidateServerCertificate(requestContext);
+
             return new ServiceFabricManagedIdentitySource(requestContext, endpointUri, identityHeader);
         }
 
+        private static void ValidateServerCertificate(RequestContext requestContext)
+        {
+#if !NET462
+            requestContext.Logger.Verbose(() => "[Managed Identity] Updating the http client to validate the server certificate.");
+
+            HttpClientHandler handler = new HttpClientHandler();
+            handler.ServerCertificateCustomValidationCallback = (message, certificate, chain, sslPolicyErrors) =>
+            {
+                if (sslPolicyErrors != System.Net.Security.SslPolicyErrors.None)
+                {
+                    return 0 == string.Compare(certificate.Thumbprint, EnvironmentVariables.IdentityServerThumbprint, StringComparison.OrdinalIgnoreCase);
+                }
+                return true;
+            };
+
+            requestContext.ServiceBundle.HttpManager.HttpClientHandler = handler;
```

Comment:
But then all the other flows will need to set ServiceBundle.HttpManager.HttpClientHandler to null and reset the HttpManager's static cache. 

I think this will be quite messy. 

I'd go for either: 

- make HttpManager immutable, e.g. by adding the special handler on the `RequestAsync` method instead of making it a property
- or get away completely from this and create a new HttpClient (and cache it) here statically